processors/base/helpers.py

```diff
@@ -203,9 +204,12 @@ def find_trial_by_identifiers(conn, identifiers):
     for source, identifier in identifiers.items():
         query = QUERY % json.dumps({source: identifier})
         records = list(conn['database'].query(query))
-        if not records:
-            continue
-        trial = conn['database']['trials'].find_one(id=records[0]['trial_id'].hex)
+        for record in records:
+            if (ignore_record_id and record['id'].hex == uuid.UUID(ignore_record_id).hex):
+                continue
+            trial = conn['database']['trials'].find_one(id=record['trial_id'].hex)
+            if trial:
```

Comment:
Good catch, this function a good candidate for a refactoring (just don't have time for it now at all). But we should be very careful here because it's core of all dedup system.